### PR TITLE
Add selection sort

### PR DESCRIPTION
**Pull request by thepulkitagarwal**
I added the code for selection sort

### CODE REVIEW

**Comment by Dawny33** [comment]
Can you pl consider making the following changes:

Thank you for contributing.
1. Renaming the function name from `selectionsort!` to `selectsort!`. I decided not to use the `-ion` construct for avoiding long function names and for ease of use.
2. Use the `greater than` check in the implementation, rather than the `less than` style. It was the style followed in the other algorithm implementations too (Ex: `bubblesort!`, etc), so it'd help maintain the same style.

File: src/Sorting.jl [hunk 1]
```diff
@@ -58,3 +58,28 @@ function bubblesort!{T}(x::AbstractArray{T})
...
+
+Sorts an array using the **Selection Sort** approach.
+"""
+function selectionsort!{T}(x::AbstractArray{T})
+
+  for i in 1:length(x)-1
+    iMin = i
+
+    for j in i+1:length(x)
+      if x[j] < x[iMin]
```
Comment (Dawny33) [inline src/Sorting.jl:74]: By `greater than` check implementation, I meant the comparison can be if  `x[j+1]>x[iMax]` instead of `x[j] < x[iMin]`.

Comment (thepulkitagarwal) [inline src/Sorting.jl:74]: So, should the algorithm find the index of the maximum element, and then sort it starting at the end of the array?

Comment (Dawny33) [inline src/Sorting.jl:74]: On a second thought, this implementation looks simple and intuitive enough. 

Pl consider changing the function name, and I'll merge it :+1:

Comment (thepulkitagarwal) [inline src/Sorting.jl:74]: I changed the function name. 

Thank you!

**Pull request merged**